### PR TITLE
storagenode/nodestats: combine stats into single RPC call

### PR DESCRIPTION
**Pull request by rikysya**
What:  Nodestats combine node stats into single endpoint

Why: To reduce unnecessary network and db calls

Please describe the tests:
 - Test 1:
 - Test 2:
 
Please describe the performance impact:

## Code Review Checklist (to be filled out by reviewer)
 - [ ] Does the PR describe what changes are being made?
 - [ ] Does the PR describe why the changes are being made?
 - [ ] Does the code follow [our style guide](https://github.com/storj/docs/blob/master/code/Style.md)?
 - [ ] Does the code follow [our testing guide](https://github.com/storj/docs/blob/master/code/Testing.md)?
 - [ ] Is the PR appropriately sized? (If it could be broken into smaller PRs it should be)
 - [ ] Does the new code have enough tests? (*every* PR should have tests or justification otherwise. Bug-fix PRs especially)
 - [ ] Does the new code have enough documentation that answers "how do I use it?" and "what does it do?"? (both source documentation and [higher level](https://github.com/storj/docs), diagrams?)
 - [ ] Does any documentation need updating?
 - [ ] Do the database access patterns make sense?

### CODE REVIEW

**Comment by rikysya** [comment]
I am not sure. I don't know why, but every time I generate new `protobuf` it changes my `go.mod` and `go.sum`. May be I forgot to checkout them. But I am also not sure if that is correct behavior

**Comment by rikysya** [comment]
Also `jenkins` told me to `tidy` my `go.mod`, so those changes might come from it

**Comment by ifraixedes** [comment]
It's strange!

File: pkg/pb/metainfo.pb.go [hunk 1]
```diff
@@ -3,17 +3,15 @@
...
-import fmt "fmt"
-import math "math"
-import _ "github.com/gogo/protobuf/gogoproto"
-import timestamp "github.com/golang/protobuf/ptypes/timestamp"
-
-import time "time"
-
 import (
-	context "golang.org/x/net/context"
+	context "context"
```
Comment (egonelbre) [inline pkg/pb/metainfo.pb.go]: why did this file change?

Comment (rikysya) [reply pkg/pb/metainfo.pb.go]: I just use pb.gen.go

File: pkg/pb/nodestats.proto [hunk 2]
```diff
@@ -10,29 +10,23 @@ import "gogo.proto";
...
-    rpc AuditCheck(AuditCheckRequest) returns (AuditCheckResponse);
-    rpc UptimeCheck(UptimeCheckRequest) returns (UptimeCheckResponse);
+    rpc GetStats(GetStatsRequest) returns (GetStatsResponse);
     rpc DailyStorageUsage(DailyStorageUsageRequest) returns (DailyStorageUsageResponse);
 }
 
-message AuditCheckRequest {}
-
-message AuditCheckResponse {
+message CheckMetric {
```
Comment (egonelbre) [inline pkg/pb/nodestats.proto]: Is there a better name for this?

Comment (egonelbre) [reply pkg/pb/nodestats.proto]: `ReputationScore`?

File: storagenode/nodestats/service.go [hunk 3]
```diff
@@ -25,8 +25,16 @@ var (
...
+// Stats encapsulates storagenode stats retrieved from the satellite
+type Stats struct {
+	SatelliteID storj.NodeID
+
+	UptimeCheck CheckMetric
+	AuditCheck  CheckMetric
+}
+
+// CheckMetric encapsulates storagenode check metrics
+type CheckMetric struct {
```
Comment (egonelbre) [inline storagenode/nodestats/service.go]: Same here, some better name.

**Pull request merged**